### ISSUE TITLE
How you implemented BatchNorm?

### ISSUE THREAD

**Issue by korabelnikov** [williford/segmentation-caffe#1]
You posted

> One of the main benefits of using my branch is the ability to use the BatchNorm layer that is included in Caffe. The batch normalization implemented in this branch of Caffe seems to be buggy and unstable.

what is your enhansments?

**Comment by williford** [comment]
The enhancements are not mine. The new version of BatchNorm comes with Caffe. HyeonwooNoh/Caffe contains an earlier Batch Normalization (BN) implementation which appears to be buggy (or perhaps less numerically stable):
https://github.com/BVLC/caffe/pull/1965

Caffe's current implementation does, however, use more memory.
